x-pack/osquerybeat/beater/osquerybeat.go

```diff
@@ -159,7 +160,7 @@ func (bt *osquerybeat) Run(b *beat.Beat) error {
 	}
 
 	// Connect publisher
-	bt.client, err = b.Publisher.Connect()
+	processors, err := bt.reconnectPublisher(b, bt.config.Inputs)
```

Comment:
the client should always be closed on shutdown.

The reason we have a `Connect` is that processors might be stateful. If possible a `beat.Client` must not be shared between multiple inputs, but each input should have its own instance.

Reply:
it is closed on shutdown. currently it's is shared. the osquerybeat was not aware or using processors before this change, something to address properly for 7.14.
from osquery perspective the input/streams is just a query groups/ queries.
this was the way the initial policy/integration structure was adopted for scheduled queries delivery.